07-complex-data-type/src/ComplexDataTypeSample.java

```diff
@@ -0,0 +1,258 @@
+        System.out.println("StoreMap using mapOf method : " + mapUsingStringMapOF);
+        System.out.println();
+
+        /**
+         * Create a map from a {@link CellSet}.
+         * Names of cells become keys, type/value pairs become values.
+         */
+
+        StoreMap mapUsingCellMapOf = StoreMap.mapOf(originalCellSet());
+        System.out.println("Map using mapOf(Cellset) method : " + mapUsingCellMapOf);
+        System.out.println();
+
+        /**
+         * A new deep-copy mutable instance can be created from any instance via the mutableCopy() method.
+         * Make deep copy of this map and list, ensuring it is mutable.
+         */
+
+        StoreList mutableList = list.mutableCopy();
+        StoreMap mutableMap = map.mutableCopy();
+
+        System.out.println("Mutable StoreList is : " + mutableList);
+        System.out.println("Mutable StoreMap is : " + mutableMap);
+
+        /**
+         * Text representation of list and map using toString() method
+         */
+
+        String listAsString = list.toString(false);
+        String mapAsString = map.toString(false);
+
+        System.out.println("StoreList in text representation : " + listAsString);
+        System.out.println("StoreMap in text representation : " + mapAsString);
+        System.out.println();
+
+        /**
+         * Construct a single StoreList and StoreMap by parser from the provided String
+         */
+
+        StoreList convertedList = StoreList.listOf(listAsString);
+        StoreMap convertedMap = StoreMap.mapOf(mapAsString);
+
+        System.out.println("StoreList parsed from String is : " + convertedList);
+        System.out.println("StoreMap parsed from String is : " + convertedMap);
+        System.out.println();
+
+        /**
+         * Creating a parser for a specific string and retrieving values of the Map.
+         *
+         */
+        ComplexDataParser parser = ComplexData.parser(mapAsString);
+        for (StoreMap p = parser.nextMap(); p != null; p = parser.nextMap()) {
+            Optional<String> name = p.find(".name").strings().first();
+            Optional<Integer> age = p.find(".age").integers().first();
+
+            // Similarly we can retrieve other values from StoreList and StoreMap.
+            System.out.println("Name retrieved by Complex data parser is : " + name.get());
+            System.out.println("Age retrieved by Complex data parser is : " + age.get());
+        }
+
+        /**
+         * Find operaions
+         * find(spec) Selection querying for efficient selecting of arbitrary data inside CDTs by type, position, and key, in various combinations
+         * get(), first(), last(), sole(), isPresent(), size()
+         */
+
+        Record<String> record = ComplexRecord.mutableRecord("rec", StoreMap.newMap().with("map", map)).toRecord();
+        System.out.println("Age is : " + Record.find(".map.age").get(0).apply(record).get().value());
```

Comment:
Why are we doing all this through functional representations? Why not just have:
```java
record.find(".map.age").get(0).map(TypedValue::value).get()
```